sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruningSuite.scala

```diff
@@ -141,6 +141,14 @@ class PruningSuite extends HiveComparisonTest with BeforeAndAfter {
       Seq("2008-04-08", "11"),
       Seq("2008-04-09", "11")))
 
+  createPruningTest("Partition pruning - with nondeterministic fields",
+    "SELECT value, hr FROM srcpart1 WHERE ds = '2008-04-08' AND rand(7) < 1",
```

Comment:
Good question. I think technically we can't push down any predicates that are placed after a non-deterministic predicate. Otherwise number of input rows may change and lead to wrong query results.